### PR TITLE
Update __cpp_concepts macro

### LINKED ISSUE #57524: `std::variant` cannot be used in constexpr functions with non trivially destructible types
Hello,

I'm having an issue with Clang 14.0.6 and GCC 12.2.0's libstdc++: initializing a `std::variant` that may contain any non trivially destructible type will trigger the following error: `placement new would change type of storage`.

Here's a [minimal repro](https://github.com/JPenuchot/bug-repros/blob/main/repros/constexpr-variant-of-non-trivially-destructible-type.cpp) as an example ([here on Compiler Explorer](https://godbolt.org/z/bq1baea5r)), which GCC 12.2.0 does accept and compile.

Regards,
Jules

### PR DESCRIPTION
**Pull request by erichkeane**
After discussion with a few others, and seeing the state of our concepts support, I believe it is worth trying to see if we can update this for Clang19.  The forcing function is that libstdc++'s `<expected>` header is guarded by this macro, so we need to update it to support that.

Closes #57524.

### CODE REVIEW

File: clang/www/cxx_status.html [hunk 1]
```diff
@@ -544,16 +544,8 @@ <h2 id="cxx20">C++20 implementation status</h2>
       </tr>
       <tr> <!-- from Cologne -->
         <td><a href="https://wg21.link/p0848r3">P0848R3</a></td>
-        <td rowspan="1" class="partial" align="center">
-          <details>
-            <summary>Clang 16 (Partial)</summary>
-            Because of other concepts implementation deficits, the __cpp_concepts macro is not yet set to 202002L.
-            Also, the related defect reports <a href="https://wg21.link/cwg1496">DR1496</a> and
-            <a href="https://wg21.link/cwg1734">DR1734</a> are not yet implemented. Accordingly, deleted
-            special member functions are treated as eligible even though they shouldn't be.
```
Comment (cor3ntin) [inline clang/www/cxx_status.html:553]: Is that not true any more?

File: clang/docs/ReleaseNotes.rst [hunk 2]
```diff
@@ -110,6 +110,11 @@ C++20 Feature Support
   templates (`P1814R0 <https://wg21.link/p1814r0>`_).
   (#GH54051).
 
+- __cpp_concepts macro now updated to `202002L`, as we are confident that,
+  modulo a handful of bugs and core issues, that our concepts implementation is
```
Comment (cor3ntin) [inline clang/docs/ReleaseNotes.rst:114]: We have fairly large hands https://github.com/llvm/llvm-project/labels/concepts - but i agree doing that early in the 19 cycle makes sense

**Review by cor3ntin — APPROVED** [review]

File: clang/docs/ReleaseNotes.rst [hunk 3]
```diff
@@ -110,6 +110,11 @@ C++20 Feature Support
...
+  sufficient. This should enable libstdc++'s `<expected>` to work correctly in
+  clang.
```
Comment (AaronBallman) [inline clang/docs/ReleaseNotes.rst:116]:
```suggestion
- We have sufficient confidence in the concepts implementation to update the
  ``__cpp_concepts`` macro to `202002L`. This enables ``<expected>`` from
  libstdc++ to work correctly with Clang.
```
??

**Review by AaronBallman — APPROVED** [review]
LGTM, feel free to take or leave my suggestion on the release note.

**Pull request merged**